[PHP 7.0] Fix bug 60471 (cherry-pick)

SammyK opened the pull request:
Sharing the love of [60471](https://bugs.php.net/bug.php?id=60471) bug fix from #2442 with PHP 7.0. :)

SammyK commented:
Hey @weltling! Is this something you'd consider pulling in before PHP 7.0 active support ends in 23 days? If so I can rebase to resolve the conflicts. If not, we can just close it out. :)

weltling commented:
@SammyK thanks for the ping. I didn't notice this before, the ancestor PR is merged 7.2+. From the code POV I'd see it acceptable. However, as the branch is closing for active support in a month or perhaps 1.5 months, i'd be yet asking @sgolemon for an opinion as she was evaluating it already. We'd have yet some time, but it'd be a bit inconvenient if some followup fix could be needed afterwards.

Thanks.

weltling commented on sapi/cli/php_cli_server.c:
```diff
@@ -2310,7 +2312,13 @@ static int php_cli_server_recv_event_read_request(php_cli_server *server, php_cl
 	if (status < 0) {
-		php_cli_server_logf("%s Invalid request (%s)", client->addr_str, errstr);
+		if (strcmp(errstr, php_cli_server_request_error_unexpected_eof) == 0 && client->parser.state == s_start_req) {
+#if PHP_DEBUG
```
This part would mean - if debug is not enabled and condition evaluates to false, no error message is shown. Is that intentional?

SammyK replied:
Yes, that is the intended behavior. :)

SammyK commented:
Oops! I just realized I had this PR pointed to `master`.  I just updated it to `PHP-7.0` and rebased for funzies.

Sounds good! The PR no longer has any merge conflicts now so if y'all wanna pull in you can. Otherwise I won't be offended if you just wanna close it. :)

weltling commented:
I've merged this as 3bc5c9f118d994dbeabd5b2ab3ae025c960a0c65 into 7.0+. Since the patch already went through QA starting at least with 7.2 alpha and the code itself comply with the BC rules. I wouldn't expect any impacts. Lets see anyway, it's like one month or a bit more yet to check until the end of active support..

Thanks!
